### ISSUE TITLE
OMERO multi-plane images

### ISSUE THREAD

**Issue by LeeKamentsky** [CellProfiler/CellProfiler#1144]
There may be a problem reading metadata in multi-plane images from OMERO. It seems that we are not correctly detecting or recording the planar configuration of a C=3, Z=10 image stack.

Balaji Ramalingam
6:57 AM (51 minutes ago)

to me, Jean-Marie 
Dear Lee,

I have shared a couple of images for your testing on the Demo Server : CP-Omero Group, under my userid.
Please check them and let me know if you can access them.

We had issues previously with sharing images, because it was a private group. Now I have changed the same to a read-annotate group, so you should definitely be able to access the images.

Please let me know if you have any trouble.
Thank you.

Best,
Balaji

---

Mr Balaji Ramalingam
Software Developer
OME Team
College of Life Sciences
University of Dundee

From: Balaji Ramalingam b.ramalingam@dundee.ac.uk
Date: Monday, 9 June 2014 17:13
To: Lee Kamentsky leek@broadinstitute.org
Cc: Jean-Marie Burel j.burel@dundee.ac.uk
Subject: Re: CP-OMERO

Dear Lee,

Many thanks for the update.
I will wait for the 2.1.1 release, and in the meanwhile make available a set of multi-dimensional images which have caused problems for us. 

I will discuss with Kenny and make these images available for your testing as well (on the demo server).
We can definitely make available a csv wrapper on OMERO if we get a clear sense of the data-model for the integration.

Either way look forward to your next release and more testing.

Best,
Balaji 

---

Mr Balaji Ramalingam
Software Developer
OME Team
College of Life Sciences
University of Dundee

From: Lee Kamentsky leek@broadinstitute.org
Date: Monday, 9 June 2014 16:20
To: Balaji Ramalingam b.ramalingam@dundee.ac.uk
Cc: Jean-Marie Burel j.burel@dundee.ac.uk
Subject: Re: CP-OMERO

Sorry about getting back to you a bit late about this - I was away for the week. There were some problems in the 2.1.0 release regarding organizing image stacks that are largely cleared-up in 2.1.1.  The 2.1.1 release should be coming out soon, perhaps within a month, and you can download a build here: http://www.cellprofiler.org/cgi-bin/trunk_build.cgi

From the screen shots, it looks like you are on the right track. It does look like something is wrong with the way we are parsing the OME-XML. There is no "C" index and consecutive frames with different channel names have increasing Z - they should have C=(0, 1, 2) and the same Z.

I can take a look at this, but I think I need to either import the source data into my database or have you grant me access to this image in order to see if 2.1.1 does a  better job and, if not, why I'm not picking up a C index for this dataset.

--Lee

On Fri, May 30, 2014 at 7:14 AM, Balaji Ramalingam b.ramalingam@dundee.ac.uk wrote:
Dear Lee,

Hope all is well at your end.
We are now trying to document the integration and workflows in our documentation.

I am stuck with handling multi-dimensional images and grouping them.

I have an omero image with the following dimensions,
T_Z_C = 1_30_3

Please see attached screen shot for the metadata extracted from file headers and the error noticed when I try to extract all images with Channel name matching ‘DAPI’.

Gist :Text search seems to be broken., I would like to group images based on their channel groupings (1 group should have 3 channels). 
But am not able to do this.

I might be doing things the wrong way as well, so please suggest a workflow which would help us handle multi-dimensional images.

And images of the same dimensionality (ZCT) seem to carry different metadata models (The ordering of the frames, Channels, etc). And this makes it difficult to construct a CSV file, with multiple multi-dimensional images , which includes metadata/Frame/Channel information.
But that can be managed if the grouping by names and types works.

Please do let me know if am not clear about a few things, we can even Skype when you get some time.
Thank you.

Best,
Balaji

---

Mr Balaji Ramalingam
Software Developer
OME Team
College of Life Sciences
University of Dundee

From: Balaji Ramalingam b.ramalingam@dundee.ac.uk
Date: Friday, 28 March 2014 17:20
To: Lee Kamentsky leek@broadinstitute.org
Cc: Jean-Marie Burel j.burel@dundee.ac.uk
Subject: Re: CP-OMERO

Wow,
This is excellent. Thank you for all the help.

I always assumed that the metadata extract from File headers, was doing it using the file names (my bad!).
Let me do the documentation and get back to you for suggestions.

Best,
Balaji

From: Lee Kamentsky leek@broadinstitute.org
Date: Friday, 28 March 2014 16:10
To: Balaji Ramalingam b.ramalingam@dundee.ac.uk
Cc: Jean-Marie Burel j.burel@dundee.ac.uk
Subject: Re: CP-OMERO

Wow - sorry this all took so long. Unfortunately, the attached CSV references a dataset in my own group, but you can change the URLs. It loads using LoadData. There's one channel in the dataset and the frame number is whatever you'd get from calling ImageReader.getIndex().

I also tried just plain dropping the URL into CellProfiler and extracting metadata from there. It actually works pretty well. You have to turn off filtering images in CellProfiler's Images module (Filter images? No Filtering) and then you have to extract metadata from the image file headers in the Metadata module and press the "Update metadata" button and then the "update" button to see the results. After that, you can filter by channel in "NamesAndTypes".

The bottom line is that it looks like it's working fantastically, you should be able to create .csv files on your side with whatever flexibility you need and even without that, users should be able to process stacks if they can manage to import the URLs into the Images module's file list.
Inline image 2
![image](https://cloud.githubusercontent.com/assets/660469/3243757/71d84bbc-f15e-11e3-9ec4-b2c3f7c2d7c7.png)
